CMSgov/design-system
# [WNMGDS-1087] Update focus styles to fix link and remove unneeded conditional focus style SCSS

## Summary
Updating the focus styles to fix an issue where multi line links were not looking good - The solution was to implement outline instead of box shadow for links only. In addition, went ahead and removed all of the conditional styling for focus styles since we are all in for focus styles now 🎉 

### Added
- Added a new mixin for focus style links
- Added more specific focus styles for HCgov secondary button

### Removed
- `$ds-include-focus-styles` conditional blocks 

### Fixed
- Fixed the multi line link focus style issue 

## How to test
Run each doc site locally to see that the focus styles look as expected and that the links use an outline as compared to everything else which uses a box shadow for the focus styles.
